lintcheck/src/config.rs

```diff
@@ -50,9 +55,12 @@ impl LintcheckConfig {
         ));
 
         // look at the --threads arg, if 0 is passed, use the threads count
-        if config.max_jobs == 0 {
-            // automatic choice
-            config.max_jobs = std::thread::available_parallelism().map_or(1, |n| n.get());
+        if config.max_jobs.is_none() {
+            config.max_jobs = Some(if config.fix || config.recursive {
+                1
+            } else {
+                std::thread::available_parallelism().map_or(1, |n| n.get())
+            });
```

Comment:
Why not make `config.max_jobs` a plain `usize` with a `#[default = 0]` and then first set it to `1` if `config.fix || config.recursive`, then if it is still zero, set it to `available_parallelism` if available, 1 otherwise?

Reply:
Indeed. Done.